### PR TITLE
coprocessor: add functions convert * to decimal

### PR DESCRIPTION
**Pull request by lonng**
## What have you changed? (mandatory)

- [x] Merge #4962 first
- [x] Merge #4987 first
- [x] Merge #5006 first

1. coprocessor: add functions convert * to decimal.

## What are the type of the changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:

- Improvement (change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

Unit tests

### CODE REVIEW

**Comment by lonng** [comment]
@brson All of the recent convert functions will serve for CAST functions (the previous CAST functions have different behavior with TiDB), this is the first step to refactor all CAST functions.

The next works:
1. The new integration test for validating the correctness of the push-down expression (new CI task can be found in https://github.com/tikv/tikv/pull/5098)
2. Correct all existing CAST expressions and implement the missings CAST functions (RPN version).
3. Open the push-down switcher of CAST expressions.

File: src/coprocessor/codec/convert.rs [hunk 1]
```diff
@@ -1312,6 +1330,45 @@ mod tests {
         assert_eq!(ctx.warnings.warnings.len(), 1);
     }
 
+    #[test]
```
[the remark this answers is not in the document]
Comment (lonng) [reply src/coprocessor/codec/convert.rs]: The major intention of this test is to guarantee the `OVERFLOWING` and `OVERFLOW_AS_WARNING` flags have been handled correctly.

**Comment by zhouqiang-cl** [comment]
@lonng seems github status has something wrong, can you commit something on this pr and the check the status

**Comment by lonng** [comment]
@zhouqiang-cl OK, I will try to push an empty commit to trigger status change.

**Pull request merged**